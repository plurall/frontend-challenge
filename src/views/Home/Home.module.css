.buttons {
  margin-top: 50px;
  display: flex;
  justify-content: center;
}

.button {
  color: #fff;
  background: var(--color-green);
  border-width: 0;
  border-style: hidden;
  border-radius: 20px;
  padding: 10px 30px;
  width: max-content;
  font-size: 14px;
  font-weight: 700;
  text-transform: uppercase;
  outline: 0;
  text-decoration: none;
  transition: all 200ms ease;
}

.button:hover {
  background-color: var(--color-dark-green);
}
