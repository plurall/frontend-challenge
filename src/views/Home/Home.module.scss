@import 'shared.module.scss';

.wrapper {
  max-width: $max-page-content-width;
  margin: 0 auto;
  padding: 2rem 1rem;
  font-family: 'Roboto', sans-serif;
  min-height: calc(100vh - var(--height-nav-bar) - var(--height-footer));

  display: flex;

  @media (max-width: $mobile-breakingpoint) {
    padding: 3rem 0.5rem;
  }
}

.content {
  margin: auto;
  height: 25rem;
  width: 100%;
  max-width: 55rem;

  display: grid;
  grid-template-columns: 1fr 18rem;
  grid-template-rows: 1fr 1fr;
  grid-template-areas:
    'image title'
    'image button';
  grid-gap: 1rem 4rem;

  @media (max-width: $mobile-breakingpoint) {
    height: initial;
    max-width: 35rem;

    display: flex;
    flex-direction: column;
    grid-gap: initial;
  }
}

.contentImage {
  grid-area: image;
  width: 100%;
  height: 100%;
  background-image: var(--image);
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;

  @media (max-width: $mobile-breakingpoint) {
    aspect-ratio: 1.5/1;
  }
}

.title {
  grid-area: title;
  font-size: 1.75rem;
  font-weight: 600;
  color: var(--color-on-background-highlight);
  align-self: flex-end;

  @media (max-width: $mobile-breakingpoint) {
    margin: 2rem 0;
    text-align: center;
  }
}

.brandImage {
  display: inline-block;
  height: 2.5rem;
  aspect-ratio: 3.33/1;
  background-image: var(--image);
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;

  transform: translateY(25%);
}

.button {
  grid-area: button;
  height: 3rem;
  width: 100%;
  align-self: center;
  padding: 0 1rem;
  font-size: 1rem;
  font-weight: 500;
  text-decoration: none;
  border-radius: 0.5rem;
  background-color: var(--color-primary);
  color: var(--color-on-primary);
  cursor: pointer;
  transition-duration: 0.2s;

  display: flex;
  align-items: center;
  justify-content: center;

  &:hover {
    filter: brightness(1.15);
  }
}
