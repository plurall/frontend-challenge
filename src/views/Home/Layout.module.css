.nav-bar {
  background-color: var(--color-dark);
  color: var(--color-white);
  font-size: 30px;
  font-weight: bolder;
  height: 400px;
  padding: 30px;
  margin-bottom: 60px;
  border-radius: 0 0 150px 0;
  display: flex;
  flex-direction: column;
  margin-top: 100px;
}

.profile-box {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.profile-picture {
  border-radius: 50%;
  width: 150px;
  height: 150px;
  margin: 10px;
}

.user-email {
  font-size: 20px;
  font-weight: 100;
  padding-bottom: 20px;
}

.user-box-btn {
  padding: 10px 20px;
  background-color: var(--color-green);
  border: none;
  border-radius: 10px;
  font-size: 20px;
  letter-spacing: 1px;
  font-weight: bolder;
  color: var(--color-white);
  transition: 0.2s;
  cursor: pointer;
  position: relative;
}

.user-box-btn:hover {
  background-color: var(--color-red);
  border-radius: 20px;
}

.loading-element {
  border-top: 5px solid var(--color-white);
  border-radius: 50%;
  width: 100px;
  height: 100px;
  animation: spinner 1s infinite linear;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(0%, 0%);
}

.transition-in {
  animation: fade linear 0.3s;
}

@keyframes fade {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes spinner {
  from {
    transform: rotate(0);
  }

  to {
    transform: rotate(180);
  }
}
