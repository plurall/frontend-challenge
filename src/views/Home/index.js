/* eslint-disable linebreak-style */
export { default } from './Home'
