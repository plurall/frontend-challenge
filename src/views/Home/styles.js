import styled from 'styled-components';

export const StyleButtonStart = styled.button`
  padding: 20px 40px;
  border-radius: 38px;
  font-size: 1.5rem;
  background: #655aa3;
  border: none;
  color: white;
  box-shadow: 1px 1px 14px #777777;
  font-weight: 600;
  cursor: pointer;
  transition: background-color 150ms ease-in;

  :hover {
    background: #39335a;
  }
`;