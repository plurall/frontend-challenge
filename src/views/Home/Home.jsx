import React from 'react'

import { SubHeader } from 'components'
import { Button } from 'plurall-ui'
import { A } from 'hookrouter'

import styles from './Home.module.css'

const Home = () => {
  return (
    <>
      <SubHeader
        breadcrumb={[{ text: 'Home' }]}
      />
      <div className={styles.wrapper}>
        <h1 className={styles.title}>Hi Welcome our Plurall music platform!! Choose your favorite artist and enjoy it .</h1>
        <A href = {'/search'}>
          <Button className={styles.btnSearch}>
            Search your favorite artist
          </Button>
        </A>  
      </div>
    </>
  )
}

export default Home
