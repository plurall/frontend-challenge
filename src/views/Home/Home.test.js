import React from 'react'
import { shallow } from 'enzyme'

import Home from './index'

describe('Home', () => {
  it('should match its snapshot', () => {
    const tree = shallow(<Home />)

    expect(tree).toMatchSnapshot()
  })
})
