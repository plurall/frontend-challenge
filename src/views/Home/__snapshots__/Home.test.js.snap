// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match its snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <SubHeader
          breadcrumb={
            Array [
              Object {
                "text": "Home",
              },
            ]
          }
          heading="Somos Front-end Challenge"
        />,
        <div
          className="wrapper"
        >
          <h2
            className="title"
          >
            Seja bem-vindo
          </h2>
          <LinkButton
            buttonHref="/search"
          >
            Buscar artista
          </LinkButton>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "breadcrumb": Array [
            Object {
              "text": "Home",
            },
          ],
          "heading": "Somos Front-end Challenge",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="title"
            >
              Seja bem-vindo
            </h2>,
            <LinkButton
              buttonHref="/search"
            >
              Buscar artista
            </LinkButton>,
          ],
          "className": "wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Seja bem-vindo",
              "className": "title",
            },
            "ref": null,
            "rendered": "Seja bem-vindo",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "buttonHref": "/search",
              "children": "Buscar artista",
            },
            "ref": null,
            "rendered": "Buscar artista",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <SubHeader
            breadcrumb={
              Array [
                Object {
                  "text": "Home",
                },
              ]
            }
            heading="Somos Front-end Challenge"
          />,
          <div
            className="wrapper"
          >
            <h2
              className="title"
            >
              Seja bem-vindo
            </h2>
            <LinkButton
              buttonHref="/search"
            >
              Buscar artista
            </LinkButton>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "breadcrumb": Array [
              Object {
                "text": "Home",
              },
            ],
            "heading": "Somos Front-end Challenge",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="title"
              >
                Seja bem-vindo
              </h2>,
              <LinkButton
                buttonHref="/search"
              >
                Buscar artista
              </LinkButton>,
            ],
            "className": "wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Seja bem-vindo",
                "className": "title",
              },
              "ref": null,
              "rendered": "Seja bem-vindo",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "buttonHref": "/search",
                "children": "Buscar artista",
              },
              "ref": null,
              "rendered": "Buscar artista",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
